src/main/java/com/synopsys/integration/alert/channel/jira/JiraIssueHandler.java

```diff
@@ -106,59 +105,38 @@ public JiraMessageResult createOrUpdateIssues(JiraIssueConfig jiraIssueConfig, M
     private Set<String> createOrUpdateIssuesPerComponent(String providerName, LinkableItem topic, Optional<LinkableItem> subTopic, Collection<ComponentItem> componentItems, JiraIssueConfig jiraIssueConfig) throws IntegrationException {
         Set<String> issueKeys = new HashSet<>();
         SetMap<String, String> missingTransitionToIssues = new SetMap<>();
-
-        ProjectComponent jiraProject = jiraIssueConfig.getProjectComponent();
-        String jiraProjectId = jiraProject.getId();
-        String jiraProjectName = jiraProject.getName();
-        String jiraProjectKey = jiraProject.getKey();
+        String jiraProjectName = jiraIssueConfig.getProjectComponent().getName();
 
         Map<String, List<ComponentItem>> combinedItemsMap = combineComponentItems(componentItems);
         for (List<ComponentItem> combinedItems : combinedItemsMap.values()) {
             try {
-                final ComponentItem arbitraryItem = combinedItems
-                                                        .stream()
-                                                        .findAny()
-                                                        .orElseThrow(() -> new AlertException("Unable to successfully combine component items for Jira Cloud issue handling."));
-                final ItemOperation operation = arbitraryItem.getOperation();
-                final String trackingKey = createAdditionalTrackingKey(arbitraryItem);
-                final IssueRequestModelFieldsBuilder fieldsBuilder = createFieldsBuilder(arbitraryItem, combinedItems, topic, subTopic, providerName);
-
-                final Optional<IssueComponent> existingIssueComponent = retrieveExistingIssue(jiraProjectKey, providerName, topic, subTopic, arbitraryItem, trackingKey);
+                ComponentItem arbitraryItem = combinedItems
+                                                  .stream()
+                                                  .findAny()
+                                                  .orElseThrow(() -> new AlertException("Unable to successfully combine component items for Jira Cloud issue handling."));
+                ItemOperation operation = arbitraryItem.getOperation();
+                String trackingKey = createAdditionalTrackingKey(arbitraryItem);
+
+                List<IssueComponent> existingIssues;
+                if (BlackDuckProjectVersionCollector.CATEGORY_TYPE.equals(arbitraryItem.getCategory())) {
+                    existingIssues = retrieveExistingIssues(jiraIssueConfig.getProjectComponent().getKey(), providerName, topic, subTopic, null, null);
+                } else {
+                    existingIssues = retrieveExistingIssues(jiraIssueConfig.getProjectComponent().getKey(), providerName, topic, subTopic, arbitraryItem, trackingKey);
```

Comment:
An alternative is to set arbitratyItem and trackingKey to null if you the check passes and just call the retrieveExistingIssues method a single time.